src/services/item/plugins/tag/ItemTag.repository.ts

```diff
@@ -24,6 +25,38 @@ export class ItemTagRepository extends AbstractRepository<ItemTag> {
     return itemTags.map(({ tag }) => tag);
   }
 
+  async getCountBy({
+    search,
+    category,
+  }: {
+    search: string;
+    category?: TagCategory;
+  }): Promise<TagCount[]> {
+    if (!search) {
+      throw new IllegalArgumentException(`search is invalid: "${search}"`);
+    }
+
+    const q = this.repository
+      .createQueryBuilder('itemTag')
+      .select(['t.name AS name', 't.category AS category', 'count(t.id) as count'])
```

Comment:
I didn't remember but are the alias required in TypeORM for the columns name?

Also, concerning the parsing of the count from string to number, you could try this (it works in pure sql):

```suggestion
      .select(['t.name AS name', 't.category AS category', 'count(t.id)::integer AS count'])
```